src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -473,8 +463,25 @@ pub fn std_cargo(builder: &Builder<'_>, target: TargetSelection, stage: u32, car
         cargo.env("MACOSX_DEPLOYMENT_TARGET", target);
     }
 
+    // Paths needed by `library/profiler_builtins/build.rs`.
     if let Some(path) = builder.config.profiler_path(target) {
         cargo.env("LLVM_PROFILER_RT_LIB", path);
+    } else if builder.config.profiler_enabled(target) {
+        let compiler_rt = if builder.config.llvm_from_ci {
```

Comment:
The `compiler-rt` directory was added to the `ci-llvm` archives only very recently. Since `ci-llvm` is used quite a lot I imagine, people who check out old branches without also rebasing on the latest `master` will have an issue with the assert on line 480.

Could you please modify this so that the directory from the CI archive is used only when the directory is actually present in the archive? We can leave a `// FIXME` to put the assert back e.g. in 6 months.

Reply:
If people are on a branch that's newer than the merge of this PR, then the timestamp mechanism should ensure that bootstrap has downloaded a sufficiently-new copy of `ci-llvm`.

And if they're on an older branch, then they'll also have the older copy of this code, without this assertion. (And nothing we write here matters, since they're running the version of bootstrap that exists on that branch!)

So I'm struggling to imagine a branch-juggling scenario where removing this assertion is helpful.

Reply:
Sorry, seems like you're right. I wrote this remark because something similar happened to me with `download-rustc` when I was checking your PR, but that probably uses a slightly different mechanism (I think that it reused an old ci-rustc archive with newer code, which causes some issue). But I haven't encountered this situation with `ci-llvm`.

Nevermind!